sympy/polys/tests/test_polytools.py

```diff
@@ -2881,7 +2882,7 @@ def test_cancel():
     M = MatrixSymbol('M', 5, 5)
     assert cancel(M[0,0] + 7) == M[0,0] + 7
     expr = sin(M[1, 4] + M[2, 1] * 5 * M[4, 0]) - 5 * M[1, 2] / z
-    assert cancel(expr) == expr
+    assert simplify(cancel(expr)) == expr
```

Comment:
This test seems wrong. The point of the function is to test `cancel`. What does this do without `simplify` here?